plugins/src/test/java/opengrok/auth/plugin/LdapServerTest.java

```diff
@@ -80,49 +80,50 @@ public void testSetGetUsername() {
     @Test
     public void testIsReachable() throws IOException, InterruptedException, URISyntaxException {
         // Start simple TCP server on test port.
-        InetAddress localhostAddr = InetAddress.getLocalHost();
-        ServerSocket serverSocket = new ServerSocket(0, 1, localhostAddr);
-        Thread thread = new Thread(() -> {
-            try {
-                while (true) {
-                    Socket client = serverSocket.accept();
-                    client.close();
+        InetAddress localhostAddr = InetAddress.getLoopbackAddress();
+        try (ServerSocket serverSocket = new ServerSocket(0, 1)) {
+            Thread thread = new Thread(() -> {
+                try {
+                    while (true) {
+                        Socket client = serverSocket.accept();
+                        client.close();
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            });
+
+            int testPort = serverSocket.getLocalPort();
+            thread.start();
+            Socket socket = null;
+            for (int i = 0; i < 3; i++) {
+                try {
+                    socket = new Socket(localhostAddr, testPort);
```

Comment:
There is a resource leak as we are not closing the first 2 sockets and also the waits make the test flaky and execute for too long. I believe the purpose is to make it work for now and we can revisit later.

Reply:
filed #3546